packages/editor/src/components/post-taxonomies/hierarchical-term-selector.js

```diff
@@ -14,7 +14,8 @@ import {
 	TextControl,
 	TreeSelect,
 	withFilters,
-	__experimentalVStack as VStack,
+	Flex,
```

Comment:
When adding the prop to `TreeSelect`, a visible amount of space is lost below the 'Add new category' submit button.


| Before prop  | After prop |
| ------------- | ------------- |
|  <img width="271" alt="Screenshot 2023-01-27 at 3 44 26 PM" src="https://user-images.githubusercontent.com/35543432/215226214-51e265c2-7e15-43ed-a31a-8c27e44cf485.png"> | <img width="273" alt="Screenshot 2023-01-27 at 3 44 02 PM" src="https://user-images.githubusercontent.com/35543432/215226229-a32daf92-9607-4109-adb3-35aa31c4f70c.png">  |

We could separate [the CSS selectors for the add and submit buttons](https://github.com/WordPress/gutenberg/blob/6a64cdfee23442ee201ea5039ae5c833d1946428/packages/editor/src/components/post-taxonomies/style.scss#L21-L24) or remove the CSS and use components instead. In this case, it worked better to use `Flex` over `VStack` so we could use `FlexItem` to ensure that the buttons retain their width, both for their size and their focus outlines. 

If the goal is to use VStack/HStack over Flex/FlexItem/FlexBlock in the future, I wonder if it would be worth adding a way to self-align the children? 🤔

Reply:
What do you think about just using `FlexItem` inside VStack/HStack? Too weird?

Reply:
Is it okay to do that? I assumed it wouldn't be since the docs mention using FlexItem/Block inside Flex. But if it's not a faux pas then I could do that. 😄

Reply:
I mean, it technically works but it is definitely debatable whether it should be a faux pas! VStack/HStack are just `Flex` components under the hood with some preset props for convenience. The perceived ickiness of using something called a "FlexItem" inside a "VStack" is what we often call a _leaky abstraction_. The Stack components are intended to abstract (i.e. hide) the notion of a flexbox into something less technical, but that abstraction leaks when one is forced to confront with flex technicalities.

So yeah, I am not pressing the matter, it's a gray area 🙃

Reply:
>we often call a leaky abstraction.

Always love to learn new terms. :) 

From [your other comment](https://github.com/WordPress/gutenberg/pull/47515#discussion_r1095795419):
>I slightly lean towards using Flex for now because it will probably be easier to refactor once we are ready to move to a better spacing component 

To confirm, are you now saying to leave this as Flex to make it easier to move later?

Reply:
In that comment, I was only advocating for using `<Flex>` over a "`<div>` with handwritten flexbox CSS". 

Whether to use `Flex`/`VStack` in this case with FlexItem, I don't have a strong opinion. I'll leave it up to your preferences 😄

Reply:
Thanks for clarifying! I'll leave it as Flex, as a 'leaky abstraction' doesn't sound ideal. Also, I wouldn't want to influence others to start doing that if they came across this instance. 😅